app/controllers/api/v2/auth_source_externals_controller.rb

```diff
@@ -0,0 +1,48 @@
+module Api
+  module V2
+    class AuthSourceExternalsController < V2::BaseController
+      include Foreman::Controller::Parameters::AuthSourceExternal
+
+      before_action :find_resource, :only => %w{show update}
+
+      api :GET, "/auth_source_externals/", N_("List external authentication sources")
+      api :GET, '/locations/:location_id/auth_source_externals/',
+          N_('List external authentication sources per location')
+      api :GET, '/organizations/:organization_id/auth_source_externals/',
+          N_('List external authentication sources per organization')
+      param_group :taxonomy_scope, ::Api::V2::BaseController
+      param_group :search_and_pagination, ::Api::V2::BaseController
+
+      def index
+        @auth_source_externals = resource_scope_for_index
+      end
+
+      api :GET, "/auth_source_externals/:id/", N_("Show an external authentication source")
+      param :id, :identifier, :required => true
+
+      def show
```

Comment:
Declared but not defined?

Reply:
that's fine, https://github.com/theforeman/foreman/pull/4865/files/92509bf2ca2ae5ea6229c9cadf01850b90ef7a74#diff-8d00a4cc43156275141995eb368a3cc2R6 is the only thing this method does which is why it's empty